src/omnicore/omnicore.cpp

```diff
@@ -784,6 +784,7 @@ static int parseTransaction(bool bRPConly, const CTransaction& wtx, int nBlock,
     std::vector<std::string> script_data;
     std::vector<std::string> address_data;
     std::vector<int64_t> value_data;
+    uint16_t nulldata_vout = 0;
```

Comment:
Well, I see your point, but in my opinion it's not really a question, whether we should support multi OP_RETURN, but in this case, whether we should change the rules to not support it. It's consensus breaking.

I haven't tested it, but I'm more or less confident that the parsing works with any payload length, multi outputs and multi pushes. :)

Speaking of the latter: initially the idea was to embed the following, once enough miners upgraded to 0.12 (IIRC):

> [push][omni] [push][payload]

Currently we do:

> [push][omnipayload]

The motivation is support of SPV clients, which use bloom filters to request data from other peers. The filters can be configured to "watch out" for a piece of data, like an address, or in our case a marker. If marker and payload are merged, this isn't possible.

CC: @msgilligan (might be interesting for you)